clio-server/src/test/scala/org/broadinstitute/clio/server/dataaccess/HttpElasticsearchDAOSpec.scala

```diff
@@ -105,7 +110,45 @@ class HttpElasticsearchDAOSpec
     } yield succeed
   }
 
-  // TODO: The following tests are demo code, and should be replaced with actual DAO methods/specs.
+  it should "return the most recent document" in {
+    import HttpElasticsearchDAOSpec._
+    import com.sksamuel.elastic4s.circe._
+    import org.broadinstitute.clio.server.dataaccess.elasticsearch.Elastic4sAutoDerivation._
+    import org.broadinstitute.clio.server.dataaccess.elasticsearch._
+
+    val index =
+      new AutoElasticsearchIndex[Document]("docs-" + UUID.randomUUID())
+
+    val documents =
+      (1 to 4)
+        .map("document-" + _)
+        .map(Document(ClioUUIDGenerator.getUUID(), _))
+
+    for {
+      _ <- httpElasticsearchDAO.createOrUpdateIndex(index)
+      _ <- Future.sequence(
+        documents.map(httpElasticsearchDAO.updateMetadata(_, index))
+      )
+      document <- httpElasticsearchDAO.getMostRecentDocument(index)
+    } yield document should be(documents.last)
+  }
+
+  it should "throw an exception if no documents exist" in {
+    import HttpElasticsearchDAOSpec._
+    import com.sksamuel.elastic4s.circe._
+    import org.broadinstitute.clio.server.dataaccess.elasticsearch.Elastic4sAutoDerivation._
+    import org.broadinstitute.clio.server.dataaccess.elasticsearch._
+
+    val index =
+      new AutoElasticsearchIndex[Document]("docs-" + UUID.randomUUID())
+
+    recoverToSucceededIf[Exception] {
+      for {
+        _ <- httpElasticsearchDAO.createOrUpdateIndex(index)
+        _ <- httpElasticsearchDAO.getMostRecentDocument(index)
+      } yield ()
+    }
```

Comment:
recoverToExceptionIf would let you test the content of the exception with an explicit assertion.
```scala
    recoverToExceptionIf[Exception] {
      for {
        _ <- httpElasticsearchDAO.createOrUpdateIndex(index)
        _ <- httpElasticsearchDAO.getMostRecentDocument(index)
      } yield ()
    } map {
      x => x.getMessage should be("...")
    }
```